`crux-llvm`: Abduction tests are sensitive to CVC5 versions

If I run the abduction-related `crux-llvm` tests using `cvc5-1.0.5`, then it fails with:

<details>

```
$ cabal run test:crux-llvm-test -- -p abd
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
*** Longer running test skipped; set CI_TEST_LEVEL=1 env var to enable
crux-llvm
  clang 14.0.0-1ubuntu1
    abd-test-file-32.c
      cvc5
        loop
          recent-clang
            cvc5 1.0.5
              abd-test-file-32
                abd-test-file-32 crux run:      OK (0.27s)
                abd-test-file-32 crux output:   FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-file-32.cvc5.good)
                             and actual (test-data/golden/abd-test-file-32.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-file-32.c:11:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 3 fact(s) would entail the goal
                      F-expect>|[Crux]   * (= x #b00000000000000000000000000000001)
                      F-actual>|[Crux]   * (bvult #b00000000000000000000000001100100 x)
                      F-expect>|[Crux]   * (= x #b00000000000000000000000000000000)
                      F-actual>|[Crux]   * (bvult #b00000000000000000000000001100011 x)
                      F-expect>|[Crux]   * (bvult x #b00000000000000000000000001100011)
                      F-actual>|[Crux]   * (= #b00000000000000000000000000000000 x)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loop.recent-clang.cvc5 1.0.5.abd-test-file-32.abd-test-file-32 crux output/' to rerun this test only.
        loopmerge
          recent-clang
            cvc5 1.0.5
              abd-test-file-32
                abd-test-file-32 crux run:      OK (0.28s)
                abd-test-file-32 crux output:   FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-file-32.cvc5.good)
                             and actual (test-data/golden/abd-test-file-32.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-file-32.c:11:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 3 fact(s) would entail the goal
                      F-expect>|[Crux]   * (= x #b00000000000000000000000000000001)
                      F-actual>|[Crux]   * (bvult #b00000000000000000000000001100100 x)
                      F-expect>|[Crux]   * (= x #b00000000000000000000000000000000)
                      F-actual>|[Crux]   * (bvult #b00000000000000000000000001100011 x)
                      F-expect>|[Crux]   * (bvult x #b00000000000000000000000001100011)
                      F-actual>|[Crux]   * (= #b00000000000000000000000000000000 x)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loopmerge.recent-clang.cvc5 1.0.5.abd-test-file-32.abd-test-file-32 crux output/' to rerun this test only.
    abd-test-file-8.c
      cvc5
        loop
          recent-clang
            cvc5 1.0.5
              abd-test-file-8
                abd-test-file-8 crux run:       OK (0.31s)
                abd-test-file-8 crux output:    FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-file-8.cvc5.good)
                             and actual (test-data/golden/abd-test-file-8.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-file-8.c:11:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 3 fact(s) would entail the goal
                      F-expect>|[Crux]   * (bvult x #b00000001)
                      F-actual>|[Crux]   * (= x #b00000000)
                      F        |[Crux]   * (bvult #b01100100 x)
                      F        |[Crux]   * (= #b00000001 x)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loop.recent-clang.cvc5 1.0.5.abd-test-file-8.abd-test-file-8 crux output/' to rerun this test only.
        loopmerge
          recent-clang
            cvc5 1.0.5
              abd-test-file-8
                abd-test-file-8 crux run:       OK (0.30s)
                abd-test-file-8 crux output:    FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-file-8.cvc5.good)
                             and actual (test-data/golden/abd-test-file-8.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-file-8.c:11:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 3 fact(s) would entail the goal
                      F-expect>|[Crux]   * (bvult x #b00000001)
                      F-actual>|[Crux]   * (= x #b00000000)
                      F        |[Crux]   * (bvult #b01100100 x)
                      F        |[Crux]   * (= #b00000001 x)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loopmerge.recent-clang.cvc5 1.0.5.abd-test-file-8.abd-test-file-8 crux output/' to rerun this test only.
    abd-test-maxint-32.c
      cvc5
        loop
          recent-clang
            cvc5 1.0.5
              abd-test-maxint-32
                abd-test-maxint-32 crux run:    OK (0.26s)
                abd-test-maxint-32 crux output: FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-maxint-32.cvc5.good)
                             and actual (test-data/golden/abd-test-maxint-32.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-maxint-32.c:6:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 2 fact(s) would entail the goal
                      F-expect>|[Crux]   * (bvult x #b00000000000000000000000000000001)
                      F-actual>|[Crux]   * (= #b00000000000000000000000000000001 x)
                      F-expect>|[Crux]   * (bvult x #b11111111111111111111111111111111)
                      F-actual>|[Crux]   * (= x #b00000000000000000000000000000000)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loop.recent-clang.cvc5 1.0.5.abd-test-maxint-32.abd-test-maxint-32 crux output/' to rerun this test only.
        loopmerge
          recent-clang
            cvc5 1.0.5
              abd-test-maxint-32
                abd-test-maxint-32 crux run:    OK (0.26s)
                abd-test-maxint-32 crux output: FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-maxint-32.cvc5.good)
                             and actual (test-data/golden/abd-test-maxint-32.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-maxint-32.c:6:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 2 fact(s) would entail the goal
                      F-expect>|[Crux]   * (bvult x #b00000000000000000000000000000001)
                      F-actual>|[Crux]   * (= #b00000000000000000000000000000001 x)
                      F-expect>|[Crux]   * (bvult x #b11111111111111111111111111111111)
                      F-actual>|[Crux]   * (= x #b00000000000000000000000000000000)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loopmerge.recent-clang.cvc5 1.0.5.abd-test-maxint-32.abd-test-maxint-32 crux output/' to rerun this test only.
    abd-test-trans-8.c
      cvc5
        loop
          recent-clang
            cvc5 1.0.5
              abd-test-trans-8
                abd-test-trans-8 crux run:      OK (1.50s)
                abd-test-trans-8 crux output:   FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-trans-8.cvc5.good)
                             and actual (test-data/golden/abd-test-trans-8.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-trans-8.c:9:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 3 fact(s) would entail the goal
                      F        |[Crux]   * (= z y)
                      F        |[Crux]   * (= (bvashr x z) #b00000001)
                      F-expect>|[Crux]   * (bvult z (bvlshr x y))
                      F-actual>|[Crux]   * (= (bvor x z) y)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loop.recent-clang.cvc5 1.0.5.abd-test-trans-8.abd-test-trans-8 crux output/' to rerun this test only.
        loopmerge
          recent-clang
            cvc5 1.0.5
              abd-test-trans-8
                abd-test-trans-8 crux run:      OK (1.50s)
                abd-test-trans-8 crux output:   FAIL
                  test/Test.hs:268:
                  MISMATCH for expected (test-data/golden/abd-test-trans-8.cvc5.good)
                             and actual (test-data/golden/abd-test-trans-8.cvc5.cvc5.out) output:
                      F        |[Crux] Found counterexample for verification goal
                      F        |[Crux]   test-data/golden/abd-test-trans-8.c:9:0: error: in main
                      F        |[Crux]   crucible_assert
                      F        |[Crux] 
                      F        |[Crux]   One of the following 3 fact(s) would entail the goal
                      F        |[Crux]   * (= z y)
                      F        |[Crux]   * (= (bvashr x z) #b00000001)
                      F-expect>|[Crux]   * (bvult z (bvlshr x y))
                      F-actual>|[Crux]   * (= (bvor x z) y)
                      F        |[Crux] Overall status: Invalid.
                  
                  Use -p '/abd/&&/loopmerge.recent-clang.cvc5 1.0.5.abd-test-trans-8.abd-test-trans-8 crux output/' to rerun this test only.

8 out of 16 tests failed (4.69s)
```
</details>

This is a bit surprising, since these tests pass on CI. But that is because the CI is using `cvc5-1.0.2`, revealing that these tests are extremely sensitive to the choice of CVC5 version being used to test with.

This state of affairs doesn't seem tenable, as it means that we would potentially need different golden output for each CVC5 version that we want to support. In some cases, the reported abducts are simply in a different order, but other times, one CVC5 version might report a completely different abduct than a different CVC5 version. As a result, we can't work around this issue by simply sorting the abducts.

The best idea that I have for making this work is to have `crux-llvm`'s test suite scrub the particular abduct values from the golden output so that the choice of abducts does not affect whether the test passes or fails. This would at least ensure that all of the relevant code paths involved in the `--get-abducts` flag continue to work without needing an explosion of different golden output combinations.

--- a/crux-llvm/crux-llvm.cabal
+++ b/crux-llvm/crux-llvm.cabal
@@ -174,4 +174,6 @@ test-suite crux-llvm-test
   build-depends:
                 crux-llvm,
                 extra,
+                regex-base,
+                regex-posix,
                 versions

--- a/crux-llvm/test-data/golden/abd-test-file-32.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-file-32.cvc5.good
@@ -2,7 +2,7 @@
 [Crux]   test-data/golden/abd-test-file-32.c:11:0: error: in main
 [Crux]   crucible_assert
 [Crux] 
-[Crux]   One of the following 3 fact(s) would entail the goal
+[Crux]   One of the following 3 facts would entail the goal
 [Crux]   * <unstable SMT formula output removed>
 [Crux]   * <unstable SMT formula output removed>
 [Crux]   * <unstable SMT formula output removed>

--- a/crux-llvm/test-data/golden/abd-test-file-32.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-file-32.cvc5.good
@@ -3,7 +3,7 @@
 [Crux]   crucible_assert
 [Crux] 
 [Crux]   One of the following 3 fact(s) would entail the goal
-[Crux]   * (= x #b00000000000000000000000000000001)
-[Crux]   * (= x #b00000000000000000000000000000000)
-[Crux]   * (bvult x #b00000000000000000000000001100011)
+[Crux]   * <unstable SMT formula output removed>
+[Crux]   * <unstable SMT formula output removed>
+[Crux]   * <unstable SMT formula output removed>
 [Crux] Overall status: Invalid.

--- a/crux-llvm/test-data/golden/abd-test-file-32.pre-clang13.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-file-32.pre-clang13.cvc5.good
@@ -1,9 +1,0 @@
-[Crux] Found counterexample for verification goal
-[Crux]   test-data/golden/abd-test-file-32.c:11:0: error: in main
-[Crux]   crucible_assert
-[Crux] 
-[Crux]   One of the following 3 fact(s) would entail the goal
-[Crux]   * (bvult x #b00000000000000000000000000000001)
-[Crux]   * (bvult #b00000000000000000000000001100100 x)
-[Crux]   * (= #b00000000000000000000000000000001 x)
-[Crux] Overall status: Invalid.

--- a/crux-llvm/test-data/golden/abd-test-file-8.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-file-8.cvc5.good
@@ -2,7 +2,7 @@
 [Crux]   test-data/golden/abd-test-file-8.c:11:0: error: in main
 [Crux]   crucible_assert
 [Crux] 
-[Crux]   One of the following 3 fact(s) would entail the goal
+[Crux]   One of the following 3 facts would entail the goal
 [Crux]   * <unstable SMT formula output removed>
 [Crux]   * <unstable SMT formula output removed>
 [Crux]   * <unstable SMT formula output removed>

--- a/crux-llvm/test-data/golden/abd-test-file-8.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-file-8.cvc5.good
@@ -3,7 +3,7 @@
 [Crux]   crucible_assert
 [Crux] 
 [Crux]   One of the following 3 fact(s) would entail the goal
-[Crux]   * (bvult x #b00000001)
-[Crux]   * (bvult #b01100100 x)
-[Crux]   * (= #b00000001 x)
+[Crux]   * <unstable SMT formula output removed>
+[Crux]   * <unstable SMT formula output removed>
+[Crux]   * <unstable SMT formula output removed>
 [Crux] Overall status: Invalid.

--- a/crux-llvm/test-data/golden/abd-test-maxint-32.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-maxint-32.cvc5.good
@@ -2,7 +2,7 @@
 [Crux]   test-data/golden/abd-test-maxint-32.c:6:0: error: in main
 [Crux]   crucible_assert
 [Crux] 
-[Crux]   One of the following 2 fact(s) would entail the goal
+[Crux]   One of the following 2 facts would entail the goal
 [Crux]   * <unstable SMT formula output removed>
 [Crux]   * <unstable SMT formula output removed>
 [Crux] Overall status: Invalid.

--- a/crux-llvm/test-data/golden/abd-test-maxint-32.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-maxint-32.cvc5.good
@@ -3,6 +3,6 @@
 [Crux]   crucible_assert
 [Crux] 
 [Crux]   One of the following 2 fact(s) would entail the goal
-[Crux]   * (bvult x #b00000000000000000000000000000001)
-[Crux]   * (bvult x #b11111111111111111111111111111111)
+[Crux]   * <unstable SMT formula output removed>
+[Crux]   * <unstable SMT formula output removed>
 [Crux] Overall status: Invalid.

--- a/crux-llvm/test-data/golden/abd-test-trans-8.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-trans-8.cvc5.good
@@ -2,7 +2,7 @@
 [Crux]   test-data/golden/abd-test-trans-8.c:9:0: error: in main
 [Crux]   crucible_assert
 [Crux] 
-[Crux]   One of the following 3 fact(s) would entail the goal
+[Crux]   One of the following 3 facts would entail the goal
 [Crux]   * <unstable SMT formula output removed>
 [Crux]   * <unstable SMT formula output removed>
 [Crux]   * <unstable SMT formula output removed>

--- a/crux-llvm/test-data/golden/abd-test-trans-8.cvc5.good
+++ b/crux-llvm/test-data/golden/abd-test-trans-8.cvc5.good
@@ -3,7 +3,7 @@
 [Crux]   crucible_assert
 [Crux] 
 [Crux]   One of the following 3 fact(s) would entail the goal
-[Crux]   * (= z y)
-[Crux]   * (= (bvashr x z) #b00000001)
-[Crux]   * (bvult z (bvlshr x y))
+[Crux]   * <unstable SMT formula output removed>
+[Crux]   * <unstable SMT formula output removed>
+[Crux]   * <unstable SMT formula output removed>
 [Crux] Overall status: Invalid.

--- a/crux-llvm/test/Test.hs
+++ b/crux-llvm/test/Test.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiWayIf #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
@@ -25,7 +26,7 @@ import           System.FilePath ( (-<.>), takeFileName )
 import           System.IO
 import           System.Process ( readProcess )
 import           Text.Read ( readMaybe )
-import           Text.Regex.Base ( makeRegex, matchM )
+import           Text.Regex.Base ( makeRegex, match, matchM )
 import           Text.Regex.Posix.ByteString.Lazy ( Regex )
 
 import qualified Test.Tasty as TT
@@ -252,28 +253,36 @@ sanitize blines =
       cleanAbductOutput :: [BSC.ByteString] -> [BSC.ByteString]
       cleanAbductOutput [] = []
       cleanAbductOutput (l:ls) =
-        case matchM re l of
-          Just ( _before :: BSC.ByteString
-               , _matched :: BSC.ByteString
-               , _after :: BSC.ByteString
-               , [numBS]
-               )
-            |  Just num <- readMaybe (BSC.unpack numBS)
-            -> -- If a line matches the error message regex, then scrub the SMT
-               -- formulas from the next <num> lines of output.
-               let (before, after) = splitAt num ls in
-               l : map cleanSMTFormula before ++ cleanAbductOutput after
-          _ -> l : cleanAbductOutput ls
-        where
-          re :: Regex
-          re = makeRegex bs
+        if |  -- If a line matches the error message regex, then scrub the SMT
+              -- formulas from the next <num> lines of output.
+              match reSingle l
+           ,  l':ls' <- ls
+           -> l : cleanSMTFormula l' : cleanAbductOutput ls'
+           |  Just ( _before :: BSC.ByteString
+                   , _matched :: BSC.ByteString
+                   , _after :: BSC.ByteString
+                   , [numBS]
+                   ) <- matchM rePlural l
 
-          -- NB: This is the same error message that is printed out in crux's
+           ,  Just num <- readMaybe (BSC.unpack numBS)
+           -> let (before, after) = splitAt num ls in
+              l : map cleanSMTFormula before ++ cleanAbductOutput after
+
+              -- Otherwise, leave the line unchanged.
+           |  otherwise
+           -> l : cleanAbductOutput ls
+        where
+          reSingle, rePlural :: Regex
+          reSingle = makeRegex bsSingle
+          rePlural = makeRegex bsPlural
+
+          -- NB: These are the same error messages that is printed out in crux's
           -- Crux.FormatOut module, but with some regex-specific twists on top
-          -- of it. This code should stay in sync with any chages to that error
-          -- message.
-          bs :: BSC.ByteString
-          bs = "One of the following ([0-9]+) fact\\(s\\) would entail the goal"
+          -- of them. This code should stay in sync with any chages to those
+          -- error messages.
+          bsSingle, bsPlural :: BSC.ByteString
+          bsSingle = "The following fact would entail the goal"
+          bsPlural = "One of the following ([0-9]+) facts would entail the goal"
 
           cleanSMTFormula :: BSC.ByteString -> BSC.ByteString
           cleanSMTFormula line =

--- a/crux-llvm/test/Test.hs
+++ b/crux-llvm/test/Test.hs
@@ -25,6 +25,8 @@ import           System.FilePath ( (-<.>), takeFileName )
 import           System.IO
 import           System.Process ( readProcess )
 import           Text.Read ( readMaybe )
+import           Text.Regex.Base ( makeRegex, matchM )
+import           Text.Regex.Posix.ByteString.Lazy ( Regex )
 
 import qualified Test.Tasty as TT
 import           Test.Tasty.HUnit ( testCase, assertFailure )
@@ -240,7 +242,45 @@ sanitize blines =
                        ]
                 then cleanLinkerOutput rest
                 else l
-  in cleanLinkerOutput . removeHashedLoc <$> blines
+
+      -- The SMT formulas suggested by the --get-abducts flag are extremely
+      -- sensitive to which versions of CVC5 and LLVM you are using. To avoid
+      -- checking in a ridiculous number of expected output combinations for
+      -- --get-abducts test cases, we scrub out the SMT formulas post facto.
+      -- This ensures that all of the relevant code paths are being exercised
+      -- while maintaining a single expected output file for each test case.
+      cleanAbductOutput :: [BSC.ByteString] -> [BSC.ByteString]
+      cleanAbductOutput [] = []
+      cleanAbductOutput (l:ls) =
+        case matchM re l of
+          Just ( _before :: BSC.ByteString
+               , _matched :: BSC.ByteString
+               , _after :: BSC.ByteString
+               , [numBS]
+               )
+            |  Just num <- readMaybe (BSC.unpack numBS)
+            -> -- If a line matches the error message regex, then scrub the SMT
+               -- formulas from the next <num> lines of output.
+               let (before, after) = splitAt num ls in
+               l : map cleanSMTFormula before ++ cleanAbductOutput after
+          _ -> l : cleanAbductOutput ls
+        where
+          re :: Regex
+          re = makeRegex bs
+
+          -- NB: This is the same error message that is printed out in crux's
+          -- Crux.FormatOut module, but with some regex-specific twists on top
+          -- of it. This code should stay in sync with any chages to that error
+          -- message.
+          bs :: BSC.ByteString
+          bs = "One of the following ([0-9]+) fact\\(s\\) would entail the goal"
+
+          cleanSMTFormula :: BSC.ByteString -> BSC.ByteString
+          cleanSMTFormula line =
+            BSC.takeWhile (/= '*') line <> "* <unstable SMT formula output removed>"
+
+  in cleanAbductOutput $
+     cleanLinkerOutput . removeHashedLoc <$> blines
 
 assertBSEq :: FilePath -> FilePath -> IO ()
 assertBSEq expectedFile actualFile = do

--- a/crux/src/Crux/FormatOut.hs
+++ b/crux/src/Crux/FormatOut.hs
@@ -84,6 +84,9 @@ sayWhatFailedGoals skipIncompl showVars allGls =
                     else [] 
                  -- print abducts, if any
                  ++ if s /= [] then
+                      -- NB: If you update the contents of this error message,
+                      -- make sure to update the corresponding regex that
+                      -- checks for this in crux-llvm/test/Test.hs.
                       PP.pretty ("One of the following " ++ show (length s) ++ " fact(s) would entail the goal")
                       : (map (\x -> PP.pretty ('*' : ' ' : x)) s)
                     else [])]

--- a/crux/src/Crux/FormatOut.hs
+++ b/crux/src/Crux/FormatOut.hs
@@ -81,7 +81,7 @@ sayWhatFailedGoals skipIncompl showVars allGls =
                  -- variable events that led to this failure
                  ++ if showVars then
                       ["Symbolic variables:", PP.indent 2 (PP.vcat (ppVars evs))]
-                    else [] 
+                    else []
                  -- print abducts, if any
                  ++ if s /= [] then
                       -- NB: If you update the contents of this error message,

--- a/crux/src/Crux/FormatOut.hs
+++ b/crux/src/Crux/FormatOut.hs
@@ -84,11 +84,18 @@ sayWhatFailedGoals skipIncompl showVars allGls =
                     else []
                  -- print abducts, if any
                  ++ if s /= [] then
-                      -- NB: If you update the contents of this error message,
-                      -- make sure to update the corresponding regex that
-                      -- checks for this in crux-llvm/test/Test.hs.
-                      PP.pretty ("One of the following " ++ show (length s) ++ " fact(s) would entail the goal")
-                      : (map (\x -> PP.pretty ('*' : ' ' : x)) s)
+                      let numFacts = length s
+                          -- NB: If you update the contents of this error
+                          -- message, make sure to update the corresponding
+                          -- regexes that check for this in
+                          -- crux-llvm/test/Test.hs.
+                          herald = PP.plural
+                                     "The following fact"
+                                     ("One of the following"
+                                        PP.<+> PP.viaShow numFacts
+                                        PP.<+> "facts")
+                                     numFacts PP.<+> "would entail the goal" in
+                      herald : (map (\x -> PP.pretty ('*' : ' ' : x)) s)
                     else [])]
          | otherwise ->
            [ PP.nest 2 $ PP.vcat [ "Failed to prove verification goal", ex ] ]

